### PR TITLE
Replace isort to ruff

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Replace isort to ruff.

 

### Details

- This is a 1st partial Pull Request of  #4115 .

### CODE REVIEW

**Comment by banesullivan** [comment]
The imports should not be changing -- we should keep the same sort rule but just change the mechanism under which they are checked

File: pyproject.toml [hunk 1]
```diff
@@ -172,3 +169,14 @@ image_cache_dir = "tests/plotting/image_cache"
...
+[tool.ruff]
+target-version = "py38"
+select = [
+    # isort
+    "I",
+]
+fixable = [
+    # isort
+    "I",
+]
```
Comment (tkoyama010) [inline pyproject.toml:182]:
```suggestion
]
[tool.ruff.per-file-ignores]
"pyvista/**/__init__.py,pyvista/__init__.py" = ["I"]
```

File: pyproject.toml [hunk 2]
```diff
@@ -101,14 +101,11 @@ pyvista = [
     'uniform.vtk',
 ]
 
-[tool.isort]
-profile = 'black'
-line_length = 100
```
Comment (tkoyama010) [inline pyproject.toml:106]: Wee need `line_length` configuration and it is discussed in https://github.com/charliermarsh/ruff/issues/3206 .

**Comment by MatthewFlamm** [comment]
In addition to line length, it looks like `import package` and `from package import thing` are handled differently.  I prefer the original implementation where it keeps these together.  The ruff version splits them into separate sections.

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
It is still not compatible enough to replace ruff with isort. It is possible that ruff will be sufficiently compatible in the future, but until then it is not practical to maintain this PullRequest. We will close this PullRequest once it has been closed. We thank the reviewers for their cooperation.

**Pull request closed without merging**